CommonTools/RecoAlgos/plugins/PrimaryVertexSorter.h

```diff
@@ -0,0 +1,274 @@
+#ifndef CommonTools_RecoAlgos_PrimaryVertexSorter_
+#define CommonTools_RecoAlgos_PrimaryVertexSorter_
+
+// system include files
+#include <memory>
+#include <string>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/Association.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "CommonTools/RecoAlgos/interface/PrimaryVertexAssignment.h"
+#include "CommonTools/RecoAlgos/interface/PrimaryVertexSorting.h"
+
+/**\class PrimaryVertexSorter
+ * \author Andrea Rizzi
+
+*/
+
+
+template <class ParticlesCollection>
+
+class PrimaryVertexSorter : public edm::stream::EDProducer<> {
+ public:
+
+  typedef edm::Association<reco::VertexCollection> CandToVertex;
+  typedef edm::ValueMap<int> CandToVertexQuality;
+
+  typedef ParticlesCollection PFCollection;
+
+  explicit PrimaryVertexSorter(const edm::ParameterSet&);
+
+  ~PrimaryVertexSorter() {}
+
+  virtual void produce(edm::Event&, const edm::EventSetup&) override;
+
+ private:
+
+  PrimaryVertexAssignment    assignmentAlgo_;
+  PrimaryVertexSorting       sortingAlgo_;
+
+  /// Candidates to be analyzed
+  edm::EDGetTokenT<PFCollection>   tokenCandidates_;
+
+  /// vertices
+  edm::EDGetTokenT<reco::VertexCollection>   tokenVertices_;
+  edm::EDGetTokenT<edm::View<reco::Candidate> >   tokenJets_;
+
+  bool produceOriginalMapping_;
+  bool produceSortedVertices_;
+  bool producePFPileUp_;
+  bool producePFNoPileUp_;
+  int  qualityCut_;
+  bool useMET_;
+};
+
+
+
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+// #include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+
+using namespace std;
```

Comment:
headers should not have ``` using namespace ```
it's less severe here, since it's only in the plugin definition and is not included in other places